include/seastar/http/httpd.hh

```diff
@@ -179,6 +182,7 @@ public:
         }).get();
      *
      */
+    [[deprecated("use listen(socket_address addr, shared_ptr<seastar::tls::server_credentials> credentials)")]]
     void set_tls_credentials(shared_ptr<seastar::tls::server_credentials> credentials);
```

Comment:
The `shared_ptr<seastar::tls::server_credentials>` appears a lot after this PR, so it deserved a `using server_credentials_ptr` alias in the class scope